watchtower/wtdb/tower.go

```diff
@@ -62,6 +62,19 @@ func (t *Tower) AddAddress(addr net.Addr) {
 	t.Addresses = append([]net.Addr{addr}, t.Addresses...)
 }
 
+// RemoveAddress removes the given address from the tower's in-memory list of
+// addresses. If the address doesn't exist, then this will act as a NOP.
+func (t *Tower) RemoveAddress(addr net.Addr) {
+	addrStr := addr.String()
+	for i, address := range t.Addresses {
+		if address.String() != addrStr {
+			continue
+		}
+		t.Addresses = append(t.Addresses[:i], t.Addresses[i+1:]...)
```

Comment:
Any reason why we don't go ahead and make this a map at this point? This would simplify both this method and `AddAddress`.

Reply:
original rationale behind using a slice is so that iteration order is known. we could make it more randomized, but the current assumption is that the most recently supplied address should be tried first before falling back to others